BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeClient.kt

```diff
@@ -263,9 +289,30 @@ open class BraintreeClient @VisibleForTesting internal constructor(
                             data = data,
                             configuration = configuration,
                             authorization = authorization,
-                            additionalHeaders = additionalHeaders,
-                            callback = responseCallback
-                        )
+                            additionalHeaders = additionalHeaders
+                        ) { response, httpError ->
+                            response?.let {
+                                try {
+                                    val params = AnalyticsEventParams(
+                                        startTime = it.startTime,
+                                        endTime = it.endTime,
+                                        endpoint = url
+                                    )
+                                    sendAnalyticsEvent(
+                                        CoreAnalytics.apiRequestLatency,
+                                        params
+                                    )
+                                    responseCallback.onResult(it.body, null)
+                                } catch (jsonException: JSONException) {
+                                    responseCallback.onResult(null, jsonException)
+                                }
+                            } ?: httpError?.let { error ->
+                                val errorMessageFormat = "Request has failed: %s"
+                                val errorMessage = String.format(errorMessageFormat, error.message)
+                                val exception = BraintreeException(errorMessage, error)
+                                responseCallback.onResult(null, exception)
+                            }
```

Comment:
I don't think we need to handle this error. If this analytics call fails - we should still proceed

Reply:
This error comes from HttpClient.notifyErrorOnMainThread, let me know what do you think: [ee093d0] (https://github.com/braintree/braintree_android/pull/1032/commits/ee093d0f4e6042000a09d9b09bb905210f7abd18)